### ISSUE TITLE
Extremely annoying disconnects/reconnects

### ISSUE THREAD

**Issue by Yoposoc** [hexchat/hexchat#1796]
Today, I am disconnect and reconnect from servers over and over again ad infinitum. It doesn't seem like my internet connection is unstable, as everything else is working fine. I even have discord open that has been staying open the whole time. I am flooding some servers with join/leave messages so I cannot use irc at all anymore. Any clues?
Ping and traceroutes didn't reveal anything strange. I had a friend reverse traceroute me to his znc server that I am using, and that turned up blank.

**Comment by Cockie** [comment]
I've been having the same problem for the past weeks. Only happens on Windows, not on Linux. Makes hexchat completely unusable.

**Comment by ns950** [comment]
This appears to still be an issue on Windows.

**Comment by mB-PiBox** [comment]
Would help if you provide the Disconnect message/reason it gives you.
I'm using Windows 10 (on multiple machines, included a Surface Pro 4) and have no issues out of the norm.

**Comment by ns950** [comment]
Yes, my apologies, that was poorly done of me. I was in a hurry. It's not my system (I am a Linux user) but I am trying to help a friend running Windows. I will gather the info when possible and post back here a more respectable bug comment with details. 

The only detail I know is that it's connecting/disconnecting from irc.oftc.net.

**Comment by ManuelSchneid3r** [comment]
Freenode as well. I get banned due to flooding.

> SASL authentication successful
> You are banned from this server- Your irc client seems broken and is flooding lots of channels. Banned for 240 min, if in error, please contact kline@freenode.net. (2016/10/9 12.14)

**Comment by sayanara** [comment]
Yes, I'm having similar issues. On RizonBNC, AniMoe, and AnimeBytes' servers, about 15 minutes after connecting the moment I try and send a message (and not a moment before) the server sends "* Disconnected (No such device or address)", "* Disconnected (20Connection reset by peer)", and "* Disconnected (20Connection reset by peer)" respectively with no extra information provided in Raw Log.
I'm on Windows 10.0.10586, and have had this issue with both the x64 and x86 builds of Hexchat 2.12.2 with Lua and Python 2.7 and 3.5 support (I've tried both).

**Comment by Yoposoc** [comment]
Interesting to know that others have a similar issue.
It's so strange-- this problem randomly stopped for me. I have no idea what triggered it and what caused it to stop.

**Comment by ShapeShifter499** [comment]
Make sure your network is setup correctly, seems I've had issues with Freenode due to IPV6 issues on Comcast. Fixed now

**Comment by sayanara** [comment]
My problem is exclusive to hexchat. Quassel and other IRC clients don't have this issue.

**Comment by ns950** [comment]
So the problem I was looking into might also have been temporary (it stopped the flooding last week). It's not completely gone away, but now I have another misconfiguration issue which I'm looking into when I have access, so I'm not clear if there is a current problem. 

The only thing seen on the screen was : 
"Disconnected (Remote host closed socket)"

Not yet tried another IRC client, only hexchat. But it has been seen on multiple systems. 

I'll try updating this coming week with more data after investigating.

**Comment by ns950** [comment]
So the problem I was looking into has also gone away, and as that was also being experienced by a Comcast customer, I'm guessing it was likely the same issue that Comcast fixed.

**Comment by ShapeShifter499** [comment]
@ns950   Ok so that's two Comcast reports including me...  who else?

@Yoposoc Do you have Comcast?

**Comment by sayanara** [comment]
I don't use comcast - it's not even available in my country.

**Comment by ShapeShifter499** [comment]
@sayanara Do you use IPV6?   I have a running theory that what ever ISP has screwed up their IPV6 configuration leading to shoddy IRC connections.   Only affects IRC networks that have IPV6 support.

**Comment by ShapeShifter499** [comment]
@sayanara Sorry disregard the last statement, I didn't fully read that you only have issue with Hexchat. Sorry

**Comment by Yoposoc** [comment]
@ShapeShifter499 Interestingly, I used to have comcast, which wasn't an issue. I had Frontier Communications when I had this issue.
Now, I am living in a different place with comcast, I have no issues now but I did before.
I doubt it's specific to only one ISP, although the problem might be on the ISP's side.

**Comment by ns950** [comment]
Just as another data point, this seems to be a misconfiguration at the provider end. The same Windows 10 hexchat client that had stopped having the problem with Comcast, resumed having the problem during travel (while flying, I believe AT&T might have been the provider but I'm not sure). 
And is working successfully again on the local Comcast network. 

Same sequence  of continuous disconnects with "(Remote host closed the connection)" messages cycling.

**Comment by ns950** [comment]
ShapeShifter499, would you happen to have a pointer to any information regarding what Comcast fixed with IPv6?

**Comment by ShapeShifter499** [comment]
@ns950 No I don't, Comcast's IPV6 connection seems shoddy and not stable for IRC and similar kinds of data. It failed yesterday but it was ok today, I might have configure my network not to allow a IPV6 connection for IRC ports.

**Comment by watceedee** [comment]
I have the same problem now, started 2 days ago, only affects some irc networks, not all

**Comment by jwshields** [comment]
This could be related to the ping timout/no ping reply problem mentioned in other issue tickets?

**Comment by squinty204** [comment]
Running fully updated Windows 10 and Ubuntu 16.04 on separate computers and have been experiencing the same kind of error messages/disconnections (with both windows and linux hexchat versions) as stated by others here.  The same situation was experienced previously  a little while ago and then seemed to have rectified itself, but now the disconnects are back with a vengeance and due to them happening I am being temporally banned on some networks.  FWIW, the windows box goes through a commercial VPN but the disconnects still occur even if the VPN is switched off; the linux box uses a direct connect to my isp.  Again fwiw, I have used xchat and hexchat for years without any problems such as have been occurring the last few months.  :-(

**Comment by californiaimage** [comment]
I am sure it is not a connectivity issue related to the ISP. It seems a connectivity issued related to Github and where it chokes out. I see this with my larger repositories and trying to clone or push for the first time.

**Comment by watceedee** [comment]
It is not related to the ISP - I run other IRC clients that doesn't have this problem. these days i ahve completely stopped using HexChat for this very reason, and i have yet to see this problem in other clients - this is a problem in the HexChat codebase, guaranteed.

**Comment by ShapeShifter499** [comment]
I should update that I no longer seem to have this issue on the latest update of Arch Linux code

**Comment by Spectrik** [comment]
I am experiencing this issue as well...running Fedora 27.

**Comment by Ali-Shaikh** [comment]
I resolved the issue by settings file transfers to auto download without interaction

**Comment by muromizm** [comment]
Hello, I have the same issue. First connection after start operating system:

` * No ping reply for 88 seconds, disconnecting.
 * Disconnected ()`

Second try of connection is successfully.

OS: Windows 10. Ver. 1709 (x64)
Client: HexChat 2.14.1(x64)
IRC Network: irc.anarxi.st (RusNet), SSL connection.

**Comment by vgturtle127** [comment]
I have the same issue, I am running 64-bit Windows 10 fully updated. And the latest stable version of HexChat.

I was permabanned from Freenode due to the flooding issue as well, so this would be good bug to prioritize.

I use this for projects I work on, so it would be much appreciated if this issue could be fixed ASAP.

**Comment by OrangeDog** [comment]
Have the same problem (~once per hour), but my disconnect message is
> Disconnected (No such device or address)

**Comment by tobytteh** [comment]
I'm having the same issue, here. Occasionally Hexchat also hangs when opening. Windows 10 64-bit, using Hexchat 2.12.4 (staying a version behind because 2.14.0 and later [fail to render emojis (!)](https://github.com/hexchat/hexchat/issues/2138), however I believe I've had this issue using 2.14.1 as well). I'm in the UK so it definitely isn't a 'Comcast' issue for me.

**Comment by grravity** [comment]
Same issue, Win10 64bit - Hexchat v2.14.3
Insane how this still is open.

**Comment by CerebralDatabank** [comment]
Same issue here.
OS: Windows 10, version 20H2
HexChat version: 2.14.3
If I leave HexChat running overnight, multiple "disconnected" and "reconnecting"/"reconnected" messages are there the next morning.

**Comment by ignitionnight** [comment]
I experienced this problem for years with a few specific servers, I recently moved and doing so upgraded my modem for gigabit cable connection and haven't had the problem since. Perhaps there is a cause based on modems or docsis technology?

Old - Arris Surfboard SB6190 Docsis 3.0
New - Motorola MB8600 DOCSIS 3.1

**Comment by rhmcruiser** [comment]
I doubt its the ISP because I have two laptops of same configuration one is RHEL 7.9 using **xchat** and other is Fedora using **hexchat**.  BOTH use same Wifi, same ISP.  BUT Issue of disconnect and reconnect (30 secs) only occurs in the system using hexchat and NO ISSUE on  system using xchat.  

( My observation on system is hexchat disconnects (once every hour or 30 mins)  and reconnects(within 30 seconds)
